Add image module

ovekyc opened the pull request:
- Edit clearDB function
- Edit apidoc rule (include image module)
- Add Image controller
- Add image controller function get
- Add test code image controller get
- Add Image manager
- Add fetchPrefix method at ImageManager
- Add test code for fetchPrefix of ImageManager
- Add Apidoc for image-get
- Add getPrefix function at KeyUtils

ovekyc committed:
Add image module

* Edit clearDB function
* Edit apidoc rule (include image module)
* Add Image controller
* Add image controller function get
* Add test code image controller get
* Add Image manager
* Add fetchPrefix method at ImageManager
* Add test code for fetchPrefix of ImageManager
* Add Apidoc for image-get
* Add getPrefix function at KeyUtils

ovekyc commented:
- rebase
- fix lint warning

hewonjeong commented on src/server/database.js:
```diff
@@ -9,9 +9,9 @@ export const clearDB = (cb) => {
+        return cb(err);
       }
-      db.open(cb);
+      return db.open(cb);
```
good!

festinalente91 replied:
sorry I changed this code to promise based code

festinalente91 commented on src/server/images/controllers.js:
```diff
@@ -0,0 +1,61 @@
+export default {
+  get(req, res, cb) {
+    const {itemid, imageid} = req.query;
+    if (itemid) {
```
why don't you use Camelcase?

festinalente91 commented on src/server/images/controllers.js:
```diff
@@ -0,0 +1,61 @@
+            return cb(new APIError(err));
+          }
+          res.status(200).send({
+            imageUrls: urls
```
How about just using the variable name as 'imageUrls'?

festinalente91 commented on src/server/images/models.js:
```diff
@@ -0,0 +1,26 @@
+import db from '../database';
+
+export default class ImageManager {
+  static fetchPrefix(prefix, cb) {
```
why did you use static?

ovekyc replied:
static make to use method without creating an instance.
I just want to use this function without create instance That is it!

hewonjeong commented on src/server/items/controllers.js:
```diff
@@ -173,7 +173,7 @@ export default {
       createdDate: currentTime.toISOString(),
       modifiedDate: currentTime.toISOString()
-    }
+    };
```
thx...☆

hewonjeong commented on src/server/images/controllers.js:
```diff
@@ -0,0 +1,61 @@
+            data.map((key) => {
+              keys.push(key);
+            });
+            resolve();
```
We get origin image's keys in data arr using this fetchPrefix function.
We have to copy element's of data array to keys array?
I think we can just call 'resolve(data)' to send keys array to .then(keys) block.

ovekyc committed:
Edit image module (get)

Edit get module for not only url but also object
Edit test code
Edit MockS3 method
Edit apidoc

ovekyc committed:
Edit variable name

* Edit apidoc

HewonJeong commented:
LGTM

ovekyc committed:
Edit clearDB

festinalente91 commented:
LGTM

hewonjeong commented:
LGTM
